### PR TITLE
Fast work with tags - optimistic tagging - for #16

### PR DESCRIPTION
**Pull request by Rurik19**
Allow insert new tag without waiting for real saving

### CODE REVIEW

File: wallabagger/js/popup.js [hunk 1]
```diff
@@ -409,26 +386,29 @@ PopupController.prototype = {
...
     
     createTags: function(data){
+
         this.articleTags = data;
+
+        this.dirtyTags = this.dirtyTags.filter( tag => this.articleTags.filter(atag => atag.slug === tag.slug).length === 0 );
+
         this.clearTagInput();
-        return this.articleTags.map( tag => {
+        this.articleTags.concat(this.dirtyTags).map( tag => {
```
Comment (Simounet) [inline wallabagger/js/popup.js]: Why are you keeping the dirty tags here? I think the "clean" tags are enough.

Comment (Rurik19) [inline wallabagger/js/popup.js]: This is nessessary: when you add several tag quickly there are several requests and if we create only clean tag from response - all other dirty tag disappears and appears again only when their responses come. This flickering is annoyng. Tags added with this response clears from dirty at line 407

**Comment by Rurik19** [comment]
This PR is not about caseinsensitive search but about interface speed.  
But I am too lazy to make a separate PR, so I just add this commit :-)

**Comment by Simounet** [comment]
I thought it was the new feature that make this to append. My bad. It sounds great!

**Pull request merged**